### PR TITLE
feat: [PIE-6824]: Passing run sequence of parent to child execution

### PR DESCRIPTION
**Pull request by prashantsharma04**
## Passing run sequence and stage node id to child execution. Graph generation of pipeline stage during retry has been fixed too. Pipeline stage is a stage-step, hence executable response need to be copied for it. Executable response contains child execution id.
Testing: In local

ReleaseNotes summary needed? Yes

Is it new task type? No
Did you make changes to the code in a backward-compatible manner? Yes
Did you add a new class which is added in kryo serialization? - No
Will the new version of the delegate receive old version task and process? Yes
If new version of the pipeline-service receives response for the old task will it be able to handle it successfully? Yes
Did you make a change in delegate task which will be created in pipeline-services via grpc call to manager? No change
If the older version of pipeline-service creates the task parameter and calls newer cg manager, will it be okay? Yes
if the newer version of pipeline-service creates the task parameter and calls older cg manager, will it be okay? Yes
Did you remove an enum → No
Did you add an enum(That too at end of list) → No
FeatureFlag Added/Removed -> No
Verify Changes with affects any -> No

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41173)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/plancreator/PipelineStagePlanCreator.java [hunk 1]
```diff
@@ -85,9 +87,11 @@ public PlanCreationResponse createPlanForField(PlanCreationContext ctx, Pipeline
     }
     pipelineStageHelper.validateNestedChainedPipeline(childPipelineEntity.get());
 
+    String planNodeId = StrategyUtils.getSwappedPlanNodeId(ctx, stageNode.getUuid());
```
Comment (naidu-harness) [inline pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/plancreator/PipelineStagePlanCreator.java:90]: add a comment here on why are we doing this?

**Review by brijeshdhakar — COMMENTED** [review]
> Reviewed 4 of 6 files at r1, 2 of 4 files at r2, 5 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41173)* status: 12 file(s) to review. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/pms/execution/strategy/identity/IdentityNodeExecutionStrategy.java` line 135 at r3](https://reviewable.io/reviews/harness/harness-core/41173#-NIbNvsjDrtFHentdums:-NIbNvsjDrtFHentdumt:bd0mo1e) ([raw file](https://github.com/harness/harness-core/blob/09642e96d5ad8834e7e94b8388522a8546119e90/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/pms/execution/strategy/identity/IdentityNodeExecutionStrategy.java#L135)):*
> ```Java
>           nodeExecution, originalNodeExecution.getRetryIds());
> 
>       // Pipeline Stage is a leaf node. Need to set executable response which contains Child Execution Id
> ```

nit: Can we also explain why this special behaviour for pipeline stage?

**Review by prashantsharma04 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41173)* status: 12 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/plancreator/PipelineStagePlanCreator.java` line 90 at r3](https://reviewable.io/reviews/harness/harness-core/41173#-NIaXHteJffu9Lsxg7fQ-r3-90:-NIfC-Np7f5xocJbO52U:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/09642e96d5ad8834e7e94b8388522a8546119e90/pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/plancreator/PipelineStagePlanCreator.java#L90)):*
<details><summary><i>Previously, naidu-harness (Sanyasi Naidu Annepu) wrote…</i></summary><blockquote>

add a comment here on why are we doing this?
</blockquote></details>

Done.

**Review by naidu-harness — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 2 of 4 files at r2, 3 of 6 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41173)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41173)* status: :shipit: complete! ✓

**Pull request merged**